code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -0,0 +1,409 @@
+        } catch (final DataReaderException dataReaderException) {
+            Log.debug(CampaignPushConstants.LOG_TAG, SELF_TAG, "Required data not found, cannot create a AEPPushTemplate object.");
+            return;
+        }
+
+        // optional push template data
+        this.title = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.TITLE, "");
+        this.body = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.BODY, "");
+        this.sound = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.SOUND, "");
+        this.imageUrl = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.IMAGE_URL, "");
+        this.channelId = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.CHANNEL_ID, "");
+        this.actionUri = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.ACTION_URI, "");
+        this.icon = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.ICON, "");
+        this.expandedBodyText = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, "");
+        this.expandedBodyTextColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, "");
+        this.titleTextColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.TITLE_TEXT_COLOR, "");
+        this.smallIconColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.SMALL_ICON_COLOR, "");
+        this.notificationBackgroundColor = DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR, "");
+
+        try {
+            final String count = data.get(CampaignPushConstants.PushPayloadKeys.BADGE_NUMBER);
+            if (count != null) {
+                this.badgeCount = Integer.parseInt(count);
+            }
+        } catch (final NumberFormatException e) {
+            Log.debug(CampaignPushConstants.LOG_TAG, SELF_TAG, "Exception in converting notification badge count to int - %s", e.getLocalizedMessage());
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            this.notificationImportance = getNotificationImportanceFromString(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_PRIORITY));
+        } else {
+            this.notificationPriority = getNotificationPriorityFromString(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_PRIORITY));
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.notificationVisibility = getNotificationVisibilityFromString(data.get(CampaignPushConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY));
+        }
+
+        this.actionType = getActionTypeFromString(data.get(CampaignPushConstants.PushPayloadKeys.ACTION_TYPE));
+        this.actionButtons = getActionButtonsFromString(data.get(CampaignPushConstants.PushPayloadKeys.ACTION_BUTTONS));
+    }
+
+    public PushTemplateType getPushTemplateType() {
```

Comment:
Any reason these getters in this class need to be public?

Reply:
no, they were generated via android studio utility which created them with public. i've removed public from each getter.